lib/ansible/plugins/inventory/foreman.py

```diff
@@ -230,6 +233,12 @@ def _populate(self):
                 if self.get_option('want_facts'):
                     self.inventory.set_variable(host['name'], 'ansible_facts', self._get_facts(host))
 
+                hostvars = combine_vars(get_group_vars(self.inventory.hosts[host['name']].get_groups()), self.inventory.hosts[host['name']].get_vars())
```

Comment:
First to make a distinction, 'groups vars' can be interpreted several ways:
 - all vars associated with a group (includes all below, normally you need to call `VariableManager` or a series of methods on diff objects)
 - vars associated with groups in inventory sources (this is what you get from `Group.get_vars()`)
 - vars from host_group_vars 'vars plugin' that looks in group_vars/ directories for matching group names (this is what the function above tries to provide)

I'm not sure we want to mix match into every inventory plugin a way to access vars plugins data, since this is normally done AFTER all inventory sources are processed and we have all groups and hosts defined. in order to compose variables for group_vars you really don't need to do it at this stage since you can just define them in group_vars, play, anywhere and they will get lazily templated.